receiver/carbonreceiver/reporter.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package carbonreceiver
+
+import (
+	"context"
+
+	"github.com/open-telemetry/opentelemetry-collector/observability"
+	"go.opencensus.io/trace"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/carbonreceiver/transport"
+)
+
+type reporter struct {
+	name          string
+	spanName      string
+	logger        *zap.Logger
+	sugaredLogger *zap.SugaredLogger // Used for generic debug logging
+}
+
+var _ (transport.Reporter) = (*reporter)(nil)
+
+func newReporter(receiverName string, logger *zap.Logger) transport.Reporter {
+	return &reporter{
+		name:          receiverName,
+		spanName:      receiverName + ".receiver",
+		logger:        logger,
+		sugaredLogger: logger.Sugar(),
+	}
+}
+
+func (r *reporter) OnDataReceived(ctx context.Context) (context.Context, *trace.Span) {
```

Comment:
When is this function called? Please add a bit more comments to this file to make it easier to read.

Reply:
Done.